### DISCUSSION TITLE
customize yaml resources

### DISCUSSION THREAD

**Discussion by mh-zeynal** [nielsbasjes/yauaa#648] [Q&A] [answered]
Hello,
I was trying to drop default resources and add new ones(as bellow):

```
UserAgentAnalyzer uaa = UserAgentAnalyzer
                .newBuilder()
                .hideMatcherLoadStats()
                .dropDefaultResources()
                .addResources("classpath*:UserAgents/**/GoogleChrome.yaml")
                .addResources("classpath*:UserAgents/**/Windows.yaml")
                .withCache(10000)
                .build();
```
                
,but I got this error:

> Unable to merge lookup 'AppNamesMapping' into 'KnownWebviewAppNames'

Is it possible for you to give some hints about solving the problem?
I've attached full error log to this message.
Thank you!
[New Text Document.txt](https://github.com/nielsbasjes/yauaa/files/9045256/New.Text.Document.txt)

**Comment by nielsbasjes** [comment, marked as answer]
The bundled rules (which are stored in many files) MUST be seen as a single thing. You cannot pick and choose individual files, if you do it will simply not work as expected (this is what you got).

If you drop those resources you can load your own rules. This is useful in scenarios where you only have a limited and very specific set of useragent strings ( like internally in a datacenter).
